src/Util.cpp

```diff
@@ -105,20 +112,17 @@ QString readFile(const QString &path, const QString &head, MessageLogger *log, b
                 log->warn(head, QString("The file [%1] does not exist").arg(path));
             LOG_WARN(QString("The file [%1] does not exist").arg(path));
         }
-        return QString();
+        return "";
```

Comment:
I think you didn't read this?

https://github.com/cpeditor/cpeditor/blob/6565bb5c6fa5369b2b8528785ff35bc96b4073b0/src/Util.hpp#L54-L56

There seems to be no specific reason for changing this line.

Reply:
I noticed. But I think that QByteArray is the file content, and we convert it to QString by QString::fromUtf8, so I add the fromUtf8 or codec->toUnicode() to all the places used this function

Reply:
And as the return type is QByteArray, it's better to return a "" than a QString()

Reply:
Is a non-exist file distinguished from an empty file now?

Reply:
Maybe the correct choice is `QByteArray()`?

Reply:
And you should also modify the comments.

Reply:
ok, I'll do it